File: SakhaTyla.Web.Ng/src/app/worker-schedule-task-core/worker-schedule-task.model.ts
import { OrderDirection } from '../core/models/order-direction.model';
import { PageSettings } from '../core/page.model';
import { WorkerScheduleTaskFilter } from './worker-schedule-task-filter.model';

export class WorkerScheduleTask {
  id: number;
  creationDate: Date;
  modificationDate: Date;
  creationUserId: number;
  modificationUserId: number;
  workerInfoId: number;
  seconds: string;
  minutes: string;
  hours: string;
  dayOfMonth: string;
  month: string;
  dayOfWeek: string;
  year: string;

  constructor(workerInfoId: number) {
    this.workerInfoId = workerInfoId;
  }
}

export class WorkerScheduleTaskListState {
  pageSize = PageSettings.pageSize;
  pageIndex = 0;
  filter = new WorkerScheduleTaskFilter();
  orderBy?: string;
  orderDirection?: OrderDirection;

  constructor(filter?: WorkerScheduleTaskFilter) {
    this.filter = filter;
  }
}

File: SakhaTyla.Core/Requests/WorkerScheduleTasks/Models/WorkerScheduleTaskModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace SakhaTyla.Core.Requests.WorkerScheduleTasks.Models
{
    public class WorkerScheduleTaskModel
    {
        public WorkerScheduleTaskModel(int workerInfoId)
        {
            WorkerInfoId = workerInfoId;
        }

        [DisplayName("Id")]
        public int Id { get; set; }

        [DisplayName("Creation Date")]
        public DateTime CreationDate { get; set; }

        [DisplayName("Modification Date")]
        public DateTime ModificationDate { get; set; }

        [DisplayName("Creation User")]
        public int? CreationUserId { get; set; }
        public Users.Models.UserShortModel? CreationUser { get; set; }

        [DisplayName("Modification User")]
        public int? ModificationUserId { get; set; }
        public Users.Models.UserShortModel? ModificationUser { get; set; }

        [DisplayName("Worker")]
        public int WorkerInfoId { get; set; }
        public WorkerInfos.Models.WorkerInfoShortModel WorkerInfo { get; set; } = null!;

        [DisplayName("Seconds")]
        public string? Seconds { get; set; }

        [DisplayName("Minutes")]
        public string? Minutes { get; set; }

        [DisplayName("Hours")]
        public string? Hours { get; set; }

        [DisplayName("Day of Month")]
        public string? DayOfMonth { get; set; }

        [DisplayName("Month")]
        public string? Month { get; set; }

        [DisplayName("Day of Week")]
        public string? DayOfWeek { get; set; }

        [DisplayName("Year")]
        public string? Year { get; set; }
    }
}
